### PR TITLE
tidb_query_expr: fix round function with real type

### PR DESCRIPTION
**Pull request by Defined2014**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/18010

What's Changed: Follow https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html, let `round` function works like MySQL.

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
fix round function works with real type to make it behave more like MySQL and TiDB.
```

### CODE REVIEW

File: components/tidb_query_expr/src/impl_math.rs [hunk 1]
```diff
@@ -551,9 +551,12 @@ fn round_with_frac_dec(arg0: &Decimal, arg1: &Int) -> Result<Option<Decimal>> {
 pub fn round_with_frac_real(arg0: &Real, arg1: &Int) -> Result<Option<Real>> {
     let number = arg0;
     let digits = arg1;
-    let power = 10.0_f64.powi(-digits as i32);
-    let frac = *number / power;
-    Ok(Some(Real::new(frac.round() * power).unwrap()))
+    let power = 10.0_f64.powi(*digits as i32);
```
Comment (Defined2014) [inline components/tidb_query_expr/src/impl_math.rs:554]: Follow TiDB, ref https://github.com/pingcap/tidb/blob/c8ba4e3ace24bebcfc1b7008acc4e337ecb4c1e0/pkg/types/helper.go#L38

**Review by wshwsh12 — APPROVED** [review]

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**